serverless/serverless
# Add AWS x-ray support for API Gateway

## What did you implement:

Closes #5564

This changes adds support for declaratively enabling [active tracing](https://docs.aws.amazon.com/xray/latest/devguide/xray-services-apigateway.html) for apigateway



## How did you implement it:

If tracing is enabled we'll create a separate `Stage` resource. This feature is opt-in through explicit tracing configuration. When using a `Stage` resource we'll remove the `StageName` from the `Deployment` resource.

We can extend the `Stage` resource configurability later on via a dedicated `serverless.yml` section in `provider`.



## How can we verify it:



To enable this you should only have to add `provider:apiGateway.stageDescription.enableTracing`

```yml
provider:
  name: aws
  tracing:
    apiGateway: true
```

## Todos:


***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO
